app/domain/authentication/authn_jwt/configuration_jwt_generic_vendor.rb

```diff
@@ -5,10 +5,9 @@ class ConfigurationJWTGenericVendor < ConfigurationInterface
       def initialize
         @restriction_validator = Authentication::AuthnJwt::ValidateRestrictionsOneToOne
         @identity_provider_factory = Authentication::AuthnJwt::CreateIdentityProvider
-        @extract_resource_restrictions = Authentication::ResourceRestrictions::ExtractResourceRestrictions.new
-        @validate_resource_restrictions = Authentication::ResourceRestrictions::ValidateResourceRestrictions.new(
-          extract_resource_restrictions: @extract_resource_restrictions
-        )
+        @restrictions_from_annotations_class = Authentication::ResourceRestrictions::GetServiceSpecificRestrictionsFromAnnotation
```

Comment:
if you are changing this area, WDYT to initialize only the validate restrictions class in the ctor:
`@validate_resource_restrictions = Authentication::ResourceRestrictions::ValidateResourceRestrictions`

And to perform the dependencies allocations in a separate function like i did [here](https://github.com/cyberark/conjur/blob/8bf5063b40fa23420b9d23cf6b5690a066e7c599/app/domain/authentication/authn_jwt/configuration_jwt_generic_vendor.rb#L41)

Reply:
Done